### ISSUE TITLE
Cannot create a new key without an outfile

### ISSUE THREAD

**Issue by gakonst** [solana-labs/solana#17930]
#### Problem

I cannot create a new keypair w/o an outfile when following the [guide](https://docs.solana.com/cli/transfer-tokens#create-a-second-wallet-address). There seems to be some bug in the CLI logic when using stdout? It works when specifying a custom outfile directory with `solana-keygen new -o <...>`.

```bash
$ solana --version
solana-cli 1.7.1 (src:cd6e1d92; feat:3097519246)

$ solana config get
Config File: <...>
RPC URL: https://api.devnet.solana.com
WebSocket URL: wss://api.devnet.solana.com/ (computed)
Keypair Path: <...>
Commitment: confirmed

$ solana-keygen new --no-outfile
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /Users/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-2.33.3/src/app/usage.rs:475:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by steviez** [comment]
HI @gakonst, thanks for taking the time to file a bug! Someone else reported this issue https://github.com/solana-labs/solana/issues/17860, and a fix has been submitted with https://github.com/solana-labs/solana/pull/17896.

In regards to moving forward, this will be fixed in v1.7.2 . In the meantime, feel free to use v1.6.11 for keypair generation

**Comment by steviez** [comment]
Closing as a duplicate of #17860
